py3status/modules/xrandr_toggle.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""
+Toggles a monitor on or off from the status bar using xrandr.
+
+Configuration parameters:
+    cache_timeout: how often we refresh usage in seconds (default: 10s)
+    format: see placeholders below
+    format_disconnected: custom text or symbol for disconnected status
+    format_off: custom text or symbol for off status
+    format_on: custom text or symbol for on status
+    number: number of the monitor we want to query
+    position: where the monitor will be toggled (ex: left-of HDMI-1)
+    source: name of the source we want to query (ex: HDMI-2)
+
+Format of status string placeholders:
+    {screen} screen source name
+    {status} on/off status
+
+Example config:
+
+```
+xrandr_toggle {
+    # we could use the monitor number from xrandr output
+    # name = "1"
+    # or the monitor id directly
+    source = "HDMI-2"
+    position = "left-of DVI-I-1"
+    format = "[{screen}:TV32\"]"
+}
+```
+
+@author mrt-prodz
+"""
+from subprocess import call, check_output
+from re import search
+from time import time
+
+
+class Py3status:
+    # available configuration parameters
+    cache_timeout = 10
+    format = '{source}:{status}'
+    format_disconnected = 'disconnected'
+    format_on = 'on'
+    format_off = 'off'
+    number = '1'
+    position = ''
+    source = ''
+
+    def __init__(self):
+        self.mon = None
+        self.status = self._xrandr()
+
+    # return error occurs
+    def _error_response(self, color, text):
+        response = {
+            'cached_until': time() + self.cache_timeout,
+            'full_text': 'xrandr_toggle: ' + text,
+            'color': color
+        }
+        return response
+
+    # query xrandr for monitors status
+    def _xrandr(self):
```

Comment:
> This function is only called at __init__() and on_click() you should probably call it whenever get_status() is called just in case it has been changed outside of this module.

If I run `xrandr` too often it causes slow down on my configuration, mouse is freezing at each query. That's why I query during instantiation and then only after a user interaction. I could set another timer in `get_status` and refresh from time to time but here at least I cannot refresh the monitors list on each call.

It is an issue if someone unplug/plug a monitor, maybe assigning a right click event to refresh manually?

> The other crucial thing to be aware is that __init__() is called when the module is instantiated. This is before the config has been set. So you are always checking number = "1"

I'm not sure I understand you, xrandr() is not using any variable beside storing `self.status` to refresh the monitors list.